Improve explanation of calc+feedback (#268)
leanprover-community/leanprover-community.github.io: templates/extras/calc.md

  { rw H2 }
end
```
In fact, `calc A = B : H ...` in tactic mode functions exactly like a 
call to `refine (calc A = B : H ...)`.

## Getting effective feedback while using calc

To get helpful error messages, keep the calc structure even before the 
proof is complete. Use `_` as in the example above or `sorry` to stand 
for missing justifications. `sorry` will supress error messages 
entirely, while `_` will generate a guiding error message. 

If the structure of calc is incorrect (e.g., missing `:` or the 
justification after it), you may see error messages that are obscure 
and/or red squiggles that end up under a random `...`. To avoid these, 
you might first populate a skeleton proof such as:

```lean
example (A B C D : ℝ ) : A = D :=
calc A = B : sorry
...    = C : _
...    = D : sorry
```

and then fill in the `sorry` and `_` gradually.

In tactic mode calc should be terminated with a comma:
```lean
have H : A = D,
{ calc A = B : sorry
  ...    = C : sorry
  ...    = D : _, 

>   sorry
> },

Reviewer: Just to keep mathlib style:
```suggestion
  sorry },
```

```
In fact, `calc A = B : H ...` in tactic mode functions exactly like a call to `refine (calc A = B : H ...)`.

## Getting effective feedback while using calc

To get helpful error messages, keep the calc structure even before the proof is complete. Use `_` as in the 
example above or `sorry` to stand for missing justifications. `sorry` will supress error messages entirely, 

> while `_` will either generate a nice error message, or in tactic mode, a new subgoal. 

Reviewer: Under the theory that "in tactic mode" is an adverbial phrase for "generate", I think the commas are supposed to go like this:
```suggestion
while `_` will either generate a nice error message or, in tactic mode, a new subgoal. 
```

> If the structure of calc is incorrect (e.g., (missing `:` or the justification after it), you may see error 

Reviewer: ```suggestion
If the structure of calc is incorrect (e.g., missing `:` or the justification after it), you may see error 
```

> messages that are obscure, and/or red squiggles end up under a random `...`. To avoid these, you might first 

Reviewer: For parallelism with "you may see" (so "you may see error messages that are obscure" and/or "you may see red squiggles that end up under a random `...`").
```suggestion
messages that are obscure and/or red squiggles that end up under a random `...`. To avoid these, you might first 
```

populate a skeleton proof such as

```lean
example (A B C D : ℝ ) : A = D :=
calc A = B : sorry
...    = C : _
...    = D : sorry
```
(in tactic mode, 
```lean
have H : A = D,
{ calc A = B : _
  ...    = C : _
  ...    = D : _, },
```

> with a comma at the end), and then to start filling in the `sorry` and `_` after that. (Idle thought: could

Reviewer: Not really sure how to phrase it nicely, but when I first read the original version yesterday I was mildly confused because it didn't mention how tactic mode was different.
```suggestion
with a comma at the end), and then start filling in each `sorry` and `_` (or each new goal in tactic mode) after that. (Idle thought: could
```

Author: @kmill I agree this was a bit awkward and didn't find a satisfying minor adjustment, so I've opted for a larger change that separates the advice for tactic mode entirely. To my eyes the smaller chunks make the whole section easier to grok.
